src/coldstorage-subaccount/ColdStorageExecutor.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import { IERC7579Execution } from "modulekit/Accounts.sol";
+import { ERC7579ExecutorBase } from "modulekit/Modules.sol";
+
+contract ColdStorageExecutor is ERC7579ExecutorBase {
+    error UnauthorizedAccess();
+
+    mapping(address => address) private _subAccountOwner;
+
+    function executeOnSubAccount(
+        address subAccount,
+        address target,
+        uint256 value,
+        bytes calldata callData
+    )
+        external
+        payable
+    {
+        if (msg.sender != _subAccountOwner[subAccount]) {
+            revert UnauthorizedAccess();
+        }
+
+        IERC7579Execution smartAccount = IERC7579Execution(subAccount);
+        smartAccount.executeFromExecutor(target, value, callData);
+    }
+
+    function onInstall(bytes calldata data) external override {
+        address owner = abi.decode(data, (address));
```

Comment:
you could `abi.encodePacked` this and slice it out with `address(bytes20(data[0:20]))`

Reply:
yea actually should do this in all of our modules. im not doing the packing often. great shout